xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -34,467 +39,1093 @@
+
+  // Recording group
+  m_timerInfoTag->m_iRecordingGroup = m_iRecordingGroup;
 
-  setButtonLabels();
+  // Set the timer's title to the channel name if it's empty or 'New Timer'
+  if (channel && (m_strTitle.empty() || m_strTitle == g_localizeStrings.Get(19056)))
+    m_timerInfoTag->m_strTitle = channel->ChannelName();
+
+  // Update summary
+  m_timerInfoTag->UpdateSummary();
 }
 
-void CGUIDialogPVRTimerSettings::InitializeSettings()
+void CGUIDialogPVRTimerSettings::SetButtonLabels()
 {
-  CGUIDialogSettingsManualBase::InitializeSettings();
-
-  CSettingCategory *category = AddCategory("pvrtimersettings", -1);
-  if (category == NULL)
+  // timer start time
+  BaseSettingControlPtr settingControl = GetSettingControl(SETTING_TMR_BEGIN);
+  if (settingControl != NULL && settingControl->GetControl() != NULL)
   {
-    CLog::Log(LOGERROR, "CGUIDialogPVRTimerSettings: unable to setup settings");
-    return;
+    if (!m_bIsNewTimer && m_bStartAnytime)
+      SET_CONTROL_LABEL2(settingControl->GetID(), g_localizeStrings.Get(19161)); // "any time"
+    else
+      SET_CONTROL_LABEL2(settingControl->GetID(), m_timerStartTimeStr);
   }
 
-  CSettingGroup *group = AddGroup(category);
-  if (group == NULL)
+  // timer end time
+  settingControl = GetSettingControl(SETTING_TMR_END);
+  if (settingControl != NULL && settingControl->GetControl() != NULL)
   {
-    CLog::Log(LOGERROR, "CGUIDialogPVRTimerSettings: unable to setup settings");
-    return;
+    if (!m_bIsNewTimer && m_bEndAnytime)
+      SET_CONTROL_LABEL2(settingControl->GetID(), g_localizeStrings.Get(19161)); // "any time"
+    else
+      SET_CONTROL_LABEL2(settingControl->GetID(), m_timerEndTimeStr);
   }
 
-  // add a condition
-  m_settingsManager->AddCondition("IsTimerDayRepeating", IsTimerDayRepeating);
+  // weekdays
+  settingControl = GetSettingControl(SETTING_TMR_WEEKDAYS);
+  if (settingControl != NULL && settingControl->GetControl() != NULL)
+    SET_CONTROL_LABEL2(settingControl->GetID(),
+                       CPVRTimerInfoTag::GetWeekdaysString(
+                        m_iWeekdays, m_timerType->IsEpgBased(), true));
+}
 
-  CPVRTimerInfoTagPtr tag = m_timerItem->GetPVRTimerInfoTag();
+void CGUIDialogPVRTimerSettings::AddCondition(
+  CSetting *setting, const std::string &identifier, SettingConditionCheck condition,
+  SettingDependencyType depType, const std::string &settingId)
+{
+  m_settingsManager->AddCondition(identifier, condition, this);
+  CSettingDependency dep(depType, m_settingsManager);
+  dep.And()->Add(
+    CSettingDependencyConditionPtr(
+      new CSettingDependencyCondition(identifier, "true", settingId, false, m_settingsManager)));
+  SettingDependencies deps(setting->GetDependencies());
+  deps.push_back(dep);
+  setting->SetDependencies(deps);
+}
 
-  m_selectedChannelEntry = 0;
-  m_channelEntries.clear();
-  m_bTimerActive = tag->IsActive();
+// static
+int CGUIDialogPVRTimerSettings::GetDateAsInt(const CDateTime &datetime)
```

Comment:
also, this conversion code shouldn't be in this class really. it's got nothing to do with the gui